hook!-full-game/src/UI/EffectsLayer.gd

```diff
@@ -0,0 +1,63 @@
+extends CanvasLayer
+
+onready var fade_panel : ColorRect = $FadePanel
+onready var vignette : TextureRect = $Vignette
+onready var tween : Tween = $Tween
+
+
+func _ready() -> void:
+	Events.connect("player_died", self, "_on_Events_player_died")
+	Events.connect("player_respawned", self, "_on_Events_player_respawned")
+
+
+func _on_Events_player_died() -> void:
+	fade_in(Defaults.DEATH_ANIMATION_TIME - Defaults.TRANSITION_TIME)
+	vignette_in(0, Defaults.DEATH_ANIMATION_TIME)
+
+
+func _on_Events_player_respawned() -> void:
+	fade_out(0, Defaults.REVIVE_ANIMATION_TIME)
+
+
+func fade_in(delay: float = 0, time: float = Defaults.TRANSITION_TIME) -> void:
```

Comment:
You could've used the animation editor for this. It's a bad practice imo to use code when you could do the job faster and better with data.

You can keep this one though, we'll likely change the animation later.

Reply:
The reasons using code here is a bad idea:

- you have an extra 60 lines that can contain bugs, or produce bugs when you change the animation
- an artist can't change the animation, they have to go through a developer to make changes
- it takes much more time to test the animation as you've got to launch the game. With the animation player, you can just press Shift D in the 2d viewport to preview the anim